packages/perseus/src/widgets/interactive-graphs/graphs/sinusoid.tsx

```diff
@@ -0,0 +1,107 @@
+import {color} from "@khanacademy/wonder-blocks-tokens";
+import {Plot} from "mafs";
+import * as React from "react";
+
+import {movePoint} from "../reducer/interactive-graph-action";
+
+import {StyledMovablePoint} from "./components/movable-point";
+
+import type {Coord} from "../../../interactive2/types";
+import type {SinusoidGraphState, MafsGraphProps} from "../types";
+import type {vec} from "mafs";
+
+type SinusoidGraphProps = MafsGraphProps<SinusoidGraphState>;
+
+export type SineCoefficient = [
+    number, // amplitude
+    number, // angularFrequency
+    number, // phase
+    number, // verticalOffset
+];
+
+export function SinusoidGraph(props: SinusoidGraphProps) {
+    const {dispatch, graphState} = props;
+
+    // Destructure the coordinates from the graph state
+    // Note: The order of the coordinates is important:
+    // The coords[0] is the root and the coords[1] is the first peak
+    const {coords} = graphState;
+
+    // Destructure the coefficients for calculating the quadratic equation
+    const [a, b, c, d] = getSinusoidCoefficients(coords);
+
+    // Move a point to a new destination
+    const handleOnMove = (destination: vec.Vector2, elementId: number) => {
+        // If the destination is invalid, we do not want to move the point
+        const validDestination = isValidDestination(
+            destination,
+            elementId,
+            coords,
+        );
```

Comment:
Question: do we need to validate the destination here, instead of in the reducer? If we did the validation in the reducer, it would be easier to write tests that assert that invalid moves really get rejected.

Reply:
Ah yes, this logic makes far more sense in the reducer -- Thank you! I've moved it over as requested, and added a test to the reducer suite. :) 

I also took a look at the quadratic validation and I think that can probably move to the reducer as well, so long as we don't mind importing the getQuadraticCoefficients function. :) I'll make a micro ticket to follow up on that.